packages/components/src/color-palette/index.native.js

```diff
@@ -95,14 +98,23 @@ function ColorPalette( {
 	const customText = __( 'Custom' );
 
 	useEffect( () => {
+		const delayedScroll = setTimeout( () => {
+			resetScrollPosition();
+		}, SEGMENTED_CONTROL_ANIMATION_DURATION_DELAY );
```

Comment:
Thank you once more for the feedback @dcalhoun 🙇 
My understanding is that the scrolling doesn't complete because when the value of `currentSegment` changes there is an extra delay before the selected segment `ScrollView` is rendered due to the `SegmentedControl` [animation](https://github.com/WordPress/gutenberg/blob/3da717b8d0ac7d7821fc6d0475695ccf3ae2829f/packages/components/src/mobile/segmented-control/index.native.js#L25). Thus the scrolling occurs on the old segment selection.
I tried various approaches (e.g. using `useLayoutEffect` instead of `useEffect`) and tweaking `SegmentedControl` implementation but nothing did the trick :(
I understand that this is more of a hack and not a proper solution. Any suggestions or hints on better approaches are more than welcome.

Reply:
Ah, OK. I see. If that is the case, my recommendation would be to rely upon the pre-existing `ANIMATION_DURATION` constant. To me, that would be much more clearer as to _what_ it is waiting on. Additionally, I do not believe we need the inline anonymous function. We could likely reference pass `resetScrollPosition` directly. 

```js
const delayedScroll = setTimeout( resetScrollPosition, ANIMATION_DURATION );
```

That said, I suppose setting a delay is an alternative approach to ["track the `contentWidth` and scroll when it changes" that was removed](https://github.com/WordPress/gutenberg/pull/30994/files/a18a14006c8221cd1a3745e88d12071b0279257f#diff-3c6feb3e2ffffa5bd97bf7af05c770db725f4dd8e1ecfa0a7bb2e0fd21427a18L181), correct? Did you find that original approach was causing issues?

Reply:
Thank you for your feedback David 🙇

> Ah, OK. I see. If that is the case, my recommendation would be to rely upon the pre-existing `ANIMATION_DURATION` constant. To me, that would communicate much clearer _what_ the function is waiting on. Additionally, I do not believe we need the inline anonymous function. We could likely reference pass `resetScrollPosition` directly.

Great points. Updated with cc018b7dc00f0b44605f89b798d9fef714473c0f

> That said, I suppose setting a delay is an alternative approach to ["track the `contentWidth` and scroll when it changes" that was removed](https://github.com/WordPress/gutenberg/pull/30994/files/a18a14006c8221cd1a3745e88d12071b0279257f#diff-3c6feb3e2ffffa5bd97bf7af05c770db725f4dd8e1ecfa0a7bb2e0fd21427a18L181), correct? Did you find that original approach was causing issues?

The original approach was not causing the issue but did not help in solving it either. The content size changes still set the scroll position [directly](https://github.com/WordPress/gutenberg/pull/30994/files#diff-3c6feb3e2ffffa5bd97bf7af05c770db725f4dd8e1ecfa0a7bb2e0fd21427a18R199). I found that tracking the `contentWidth` was not necessary since it was actually used [only on Android](https://github.com/WordPress/gutenberg/pull/30994/files#diff-3c6feb3e2ffffa5bd97bf7af05c770db725f4dd8e1ecfa0a7bb2e0fd21427a18L157) and the workaround seemed to have no actual effect any longer.

Reply:
Hey @dcalhoun 👋 
I updated the PR and added some extra notes on the known issues. Both the original issues are now resolved (  
https://github.com/WordPress/gutenberg/issues/30878 and https://github.com/wordpress-mobile/gutenberg-mobile/issues/3104) and the [missing custom gradient button](https://github.com/wordpress-mobile/gutenberg-mobile/issues/3104#issuecomment-820581892) is now back on both platforms.